### PR TITLE
Provide component instance type in callbacks and methods

### PR DESCRIPTION
**Pull request by lukyanovas**
Fixes [T613738](https://www.devexpress.com/Support/Center/Question/Details/T613738)

### CODE REVIEW

**Comment by IlyaKhD** [comment]
I suggest generating interfaces, that are not derived from as non-generics.
E.g. instead of
```ts
export interface dxAccordionOptions<T = dxAccordion> extends CollectionWidgetOptions<T> {
```
generate this:
```ts
export interface dxAccordionOptions extends CollectionWidgetOptions<dxAccordion> {
```

**Review by IlyaKhD — CHANGES_REQUESTED** [review]
Generate interfaces, that are not derived from as non-generics

**Comment by lukyanovas** [comment]
@IlyaKhD 
well, let's try to generate! 😃

File: js/core/component.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ var Component = Class.inherit({
             * @name ComponentOptions.onInitialized
             * @type function
             * @type_function_param1 e:object
-            * @type_function_param1_field1 component:Component
+            * @type_function_param1_field1 component:this
```
Comment (San4es) [inline js/core/component.js:54]: The 'this' string as a type looks strange for me on the documentation comments side.

**Comment by lukyanovas** [comment]
@IlyaKhD 
Please, check the new changes.

@San4es 
It's some kind of compromise. We had 3 options:
1) to add "this" type like in TS
2) to transform all ComponentOptions to generic interfaces
3) to duplicate all callback doc-comments

We think that "this" type is less confusing of these.

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**